### ISSUE TITLE
Formatted chat not showing?

### ISSUE THREAD

**Issue by Techpro709** [PrismarineJS/mineflayer#3164]
- [ ] The [FAQ](https://github.com/PrismarineJS/mineflayer/blob/master/docs/FAQ.md) doesn't contain a resolution to my issue 

## Versions
 - mineflayer: 1.20 (I think didn't specify version)
 - server: vanilla/spigot/paper (not sure but I think 1.19.3 with viaversion (not my server, my friends))
 - node: v20.5.1 (node -v)

## Detailed description of a problem
A clear and concise description of what the problem is, with as much context as possible.
What are you building? What problem are you trying to solve?

Tvskymine.minehut.gg chat messages are returning as null when I do jsonMsg

Yet minehut lobby works just fine, im guessing its cuz of chatformatting or something.

## What did you try yet?

Did you try any method from the API?
tryied jsonMSG instead of message and username
Did you try any example? Any error from those?

## Your current code
```js

// Redirect in-game messages to Discord channel
bot.on('message', (username, message,jsonMsg) => {
  // Ignore messages from the bot itself
//   if (username === bot.username) return

  channel.send(`Chat: ${jsonMsg}`)
  console.log(`Chat: ${jsonMsg}`)
})

bot.on('whisper', (username, message,jsonMsg) => {
  // Ignore messages from the bot itself
//   if (username === bot.username) return

  channel.send(`Whisper: ${jsonMsg}`)
  console.log(`Whisper: ${jsonMsg}`)
})



```

(btw note that I have it sending to discord channel and console cuz i didn't know if it was discord problem)

## Expected behavior

I want all chat messages and server messages to show up in console / discord

(It currently only shows when ClearLag is going to clear dropped items)

## Additional context
Nope thanks. If I remember more details ill edit / add comment

**Comment by Techpro709** [comment]
Wait I just saw the thing at top with FAQ, ima test the chat-parser

**Comment by PondWader** [comment]
Use the `message` or `messagestr` events to get all messages including system messages or add your own chat pattern for the server.

**Comment by MrFiend179** [comment]
For my own botting dashboard that features colored chat i use this code
~~~js
bot.on("message", (m) => {
     console.log(`Chat: ${m.json}`)
 });
~~~

After formatting everything it looks like this
![fsadxzczx](https://github.com/PrismarineJS/mineflayer/assets/80760572/fed74715-f251-4850-b549-e44a6b6e4a5f)

**Comment by Techpro709** [comment]
Interesting, does this use the `chat-parser` module or no?

**Comment by Techpro709** [comment]
Sorry about late response, I haven't been actively developing because of school :)

**Comment by Techpro709** [comment]
Whoops accedentally closed this xd

**Comment by MrFiend179** [comment]
> Interesting, does this use the `chat-parser` module or no?

No, just built in message event, the html color and tag formatting was done by me manually and if u r interested I may be able to share the formatting code as well.
